### PR TITLE
LUCENE-10502: add entry

### PR DESCRIPTION
**Pull request by LuXugang**
Add entry.

### CODE REVIEW

File: lucene/CHANGES.txt [hunk 1]
```diff
@@ -149,6 +149,9 @@ Optimizations
   when the search order and the index order are in opposite directions.
   (Jianping Weng)
 
+* LUCENE-10502: Use IndexedDISI to store docIds and DirectMonotonicWriter/Reader to handle
+  ordToDoc in HNSW vectors (Mayya Sharipova, Julie Tibshirani, Lu Xugang)
```
Comment (mayya-sharipova) [inline lucene/CHANGES.txt:153]: You don't need to add our names (mine and Julie, as we were reviewers), and in any way please put your name first as the main contributor.

Comment (LuXugang) [reply lucene/CHANGES.txt]: Got it.

Comment (LuXugang) [reply lucene/CHANGES.txt]: @mayya-sharipova Done.

**Comment by mayya-sharipova** [comment]
@LuXugang Thanks for your work, please feel free to open PR against `branch_9x` cherry-picking this commit and also 6040d1648f6e30107086c1f9a159c1979498fb4e

**Comment by LuXugang** [comment]
> @LuXugang Thanks for your work, please feel free to open PR against `branch_9x` cherry-picking this commit and also [6040d16](https://github.com/apache/lucene/commit/6040d1648f6e30107086c1f9a159c1979498fb4e)

Hi, @mayya-sharipova see Backports https://github.com/apache/lucene/pull/885

**Pull request merged**